### ISSUE TITLE
Missing CERES_EXPORT for Context

### ISSUE THREAD

**Issue by cfneuhaus** [ceres-solver/ceres-solver#632]
The `ceres::Context` class is missing the `CERES_EXPORT` macro. This results in undefined references when using `Context::Create`.
I am using the most recent version 2.0.0.

**Comment by sandwichmaker** [comment]
I gather this is on MSVC? what is it that you are trying to do with the Context object?

**Comment by cfneuhaus** [comment]
Nope, this is gcc version 10.2.0 on Linux. I have a function that is executed quite frequently and that is re-creating new `ceres::Problems` every time it runs. A while ago, I stumbled upon Context in the [documentation](http://ceres-solver.org/nnls_modeling.html?highlight=context#_CPPv4N5ceres7Problem7Options7contextE), and it said "This may help to reduce computation time as Ceres can reuse expensive objects to create.". 
It sounded like a good idea to re-use ThreadPools so I added a statically allocated Context at that point.
Haven't benchmarked to be honest and I don't think the impact is huge, so it's not extremely important for me. Just thought I'd let you know.

**Comment by sandwichmaker** [comment]
Thanks for letting us know. @alexsmac can you take a look please?
If the creation destruction of the problem object is a performance concern, why not just have a single problem object and reuse it? or a pool of them? Also how much time are you spending in the solve compared to problem creation? unless your problems are tiny these times should not even be comparable.

**Comment by cfneuhaus** [comment]
You are completely right: The time to construct the problem almost certainly does not matter in comparison to the solving time (which is like 4-5 ms at about 1500 residuals for me; this is a real-time SLAM problem). I have simply stumbled across this as a micro-optimization while thoroughly reading the docs and added it because why not? That is all - it's not that I observed some performance issue for problem construction/destruction. As an 'actual' benefit, it hides tons of [New Thread/Thread Exited] messages in gdb ;-).
I have not considered reusing the `ceres::Problem`s because I have different residuals on every call I'd have to remove all residuals and add new ones every time - I have not looked into the details of the implementation but the interface does not make it seem like this would be a good/efficient/advisable thing to do.
